resource/Resource.md

```diff
@@ -0,0 +1,133 @@
+# Resource API Overview
+The resource library primarily defines a type that captures about information about the entity
+for which stats or traces are recorded. It further provides a framework for detection of
+resource information from the environment and progressive population as signals propagate
+from the core instrumentation library to the a backend's exporter.
+
+## Resource type
+A `Resource` describes the entity for which a signal was collected through two fields:
+* `type`: a string which describes a well-known type of entity. It SHOULD be namespaced
+to avoid collisions across different environment, e.g. `kubernetes.io/container`,
+`cloud.google.com/gce/instance`. 
+* `labels`: a dictionary of labels with string keys and values that provide information
+about the entity.
+
+Type strings and label keys MAY start with a domain name and MAY be further namespaced through
+single slashes. All other characters MUST be alphanumeric. Label values MAY contain any valid
+unicode character.
+
+Implementations MAY define a `Resource` data type, constructed from the parameters above.
+Resource MAY have getters for retrieving all the information used in `Resource` definition.
+
+Example in Go:
+```go
+type Resource {
+	Type   string
+	Labels map[string]string
```

Comment:
Your implementation had Labels as Tags. Are we going with Labels?

Reply:
Yes, we changed it from labels to tags. @ramonza asked elsewhere yesterday whether "labels" or "properties" would make more sense.  
Personally I don't feel strongly – there are arguments for either. Happy to change it to whatever you decide on.